extensions/amp-skimlinks/0.1/skimlinks.js

```diff
@@ -0,0 +1,62 @@
+import {DOM} from "./dom"
+import {skimConst} from "./const"
+
+export class Skimlinks {
+  constructor(config) {
+    
+    this.createAffiliateUrl = (url) => {
+      return url ? "https://" +
+          this.redirectingHost +
+          "/?url=" + encodeURIComponent(url) + //destination url
+          "&xs=1" + //xs=1 means skimlinks
+          "&id=" + this.skimId : ""
+    }
+    
+    this.shouldRedirect = (url) => {
+      //Url is affiliatable or unknown
+      return !this.page.isSameDomainUrl(url) && (this.page.isAffiliatableUrl(url) ||
+        (this.affiliateUnkownLinks && !this.page.isNAUrl(url)))
+    }
+    
+    this.rewriteLinkUrl = (link) => {
+      let origUrl = link.getAttribute('href')
+      let newUrl = this.createAffiliateUrl(origUrl)
+      link.href = newUrl
+      setTimeout(() => link.setAttribute("href", origUrl), 100)
```

Comment:
not sure if amp deals with IE but if so for them we need longer timeout to swap back